commit 4094af63a071cc20b53f1f41767de5f4657d421c
Author: Tim Gardner <tim.gardner@canonical.com>
Date:   Thu Mar 6 10:20:15 2014 -0700

    UBUNTU: SAUCE: (no-up) mei_me: Add module parameter to disable MSI
    
    mei_me.disable_msi = 1
    
    Signed-off-by: Tim Gardner <tim.gardner@canonical.com>

diff --git a/drivers/misc/mei/pci-me.c b/drivers/misc/mei/pci-me.c
index 5eb9b75..572ac98 100644
--- a/drivers/misc/mei/pci-me.c
+++ b/drivers/misc/mei/pci-me.c
@@ -41,6 +41,9 @@
 #include "hw-me-regs.h"
 #include "hw-me.h"
 
+static bool disable_msi;
+module_param(disable_msi, bool, 0);
+
 /* mei_pci_tbl - PCI Device ID Table */
 static const struct pci_device_id mei_me_pci_tbl[] = {
 	{MEI_PCI_DEVICE(MEI_DEV_ID_82946GZ, mei_me_legacy_cfg)},
@@ -190,7 +193,8 @@ static int mei_me_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
 		err = -ENOMEM;
 		goto free_device;
 	}
-	pci_enable_msi(pdev);
+	if (!disable_msi)
+		pci_enable_msi(pdev);
 
 	 /* request and enable interrupt */
 	irqflags = pci_dev_msi_enabled(pdev) ? IRQF_ONESHOT : IRQF_SHARED;
